modules/core/src/v2/wallet.ts

```diff
@@ -517,90 +519,87 @@ export class Wallet {
   }
 
   /**
-   * Consolidate unspents on a wallet
+   * Consolidate or fanout unspents on a wallet
+   *
+   * @param {String} routeName - either `consolidate` or `fanout`
    *
    * @param {Object} params - parameters object
+   *
+   * Wallet parameters:
    * @param {String} params.walletPassphrase - the users wallet passphrase
-   * @param {String} params.prevId - used in batch requests
-   * @param {Number} params.limit - the number of unspents retrieved per call
-   * @param {Number} params.minValue - the minimum value of unspents to use in satoshis/kB
-   * @param {Number} params.maxValue - the maximum value of unspents to use in satoshis/kB
-   * @param {Number} params.minHeight - the minimum height of unspents on the block chain to use
-   * @param {Number} params.targetUnspentPoolSize - the number of unspents you want after the consolidation of valid unspents
-   * @param {Number} params.minConfirms - all selected unspents will have at least this many confirmations
+   * @param {String} params.xprv - the private key in string form if the walletPassphrase is not available
+   *
+   * Fee parameters:
    * @param {Number} params.feeRate - The fee rate to use for the consolidation in satoshis/kB
    * @param {Number} params.maxFeeRate - upper limit for feeRate in satoshis/kB
    * @param {Number} params.maxFeePercentage - the maximum relative portion that you're willing to spend towards fees
+   * @param {Number} params.feeTxConfirmTarget - estimate the fees to aim for first confirmation with this number of blocks
+   *
+   * Input parameters:
+   * @param {Number} params.minValue - the minimum value of unspents to use in satoshis
+   * @param {Number} params.maxValue - the maximum value of unspents to use in satoshis
+   * @param {Number} params.minHeight - the minimum height of unspents on the block chain to use
+   * @param {Number} params.minConfirms - all selected unspents will have at least this many confirmations
+   * @param {Boolean} params.enforceMinConfirmsForChange - if true, minConfirms also applies to change outputs
+   * @param {Number} params.limit                for routeName === 'consolidate'
+   *                 params.maxNumUnspentsToUse  for routeName === 'fanout'
+   *                  - maximum number of unspents you want to use in the transaction
+   * Output parameters:
+   * @param {Number} params.numUnspentsToMake - the number of new unspents to make
+   *
    * @param callback
    * @returns txHex {String} the txHex of the incomplete transaction that needs to be signed by the user in the SDK
    */
-  consolidateUnspents(params: any = {}, callback?: NodeCallback<any>): Bluebird<any> {
-    const self = this;
```

Comment:
Why remove this? Accessing this via outer `self` variable allows tsc to check properties of `this`. Inside coroutines, `this` is inferred to `any` (tsc cannot know about dynamic `this` bindings at compile type).

Reply:
ah I see

`self = this` is a problem because sometimes devs accidentally use `this` anyway (in vanilla js)